components/elm/src/biogeochem/AllocationMod.F90

```diff
@@ -370,326 +356,135 @@ subroutine Allocation1_PlantNPDemand (bounds, num_soilc, filter_soilc, num_soilp
-         km_nit                       => veg_vp%km_nit                                     , &
-         km_den                       => veg_vp%km_den                                     , &
-         decompmicc_patch_vr          => veg_vp%decompmicc_patch_vr                        , &
-         slatop                       => veg_vp%slatop                                     , &
-         t_scalar                     => col_cf%t_scalar                          , &
-         w_scalar                     => col_cf%w_scalar                          , &
-         smin_nh4_to_plant_patch      => veg_nf%smin_nh4_to_plant             , &                                   
-         smin_no3_to_plant_patch      => veg_nf%smin_no3_to_plant             , &
-         sminn_to_plant_patch         => veg_nf%sminn_to_plant                , &
-         pnup_pfrootc                 => veg_ns%pnup_pfrootc                 , &
-         leafn                        => veg_ns%leafn                          &
+         benefit_pgpp_pleafc          => veg_ns%benefit_pgpp_pleafc     &
          )
 
-
-      sminp_to_plant_patch         => veg_pf%sminp_to_plant
-      secondp_vr                   => col_ps%secondp_vr
-      leafp                        => veg_ps%leafp
-      col_plant_ndemand_vr         => col_nf%col_plant_ndemand_vr
-      col_plant_nh4demand_vr       => col_nf%col_plant_nh4demand_vr
-      col_plant_no3demand_vr       => col_nf%col_plant_no3demand_vr
-      col_plant_pdemand_vr         => col_pf%col_plant_pdemand_vr
-      plant_nh4demand_vr_patch     => veg_nf%plant_nh4demand_vr
-      plant_no3demand_vr_patch     => veg_nf%plant_no3demand_vr
-      plant_ndemand_vr_patch       => veg_nf%plant_ndemand_vr
-      plant_pdemand_vr_patch       => veg_pf%plant_pdemand_vr
-      actual_immob_no3             => col_nf%actual_immob_no3
-      actual_immob_nh4             => col_nf%actual_immob_nh4
-      adsorb_to_labilep_vr         => col_pf%adsorb_to_labilep_vr
-      desorb_to_solutionp_vr       => col_pf%desorb_to_solutionp_vr
-      primp_to_labilep_vr_col      => col_pf%primp_to_labilep_vr
-      biochem_pmin_vr_col          => col_pf%biochem_pmin_vr
-      secondp_to_labilep_vr_col    => col_pf%secondp_to_labilep_vr
-      labilep_to_secondp_vr_col    => col_pf%labilep_to_secondp_vr
-      labilep_vr                   => col_ps%labilep_vr
-      benefit_pgpp_pleafc          => veg_ns%benefit_pgpp_pleafc
-
-      ! for debug
-      plant_n_uptake_flux          => col_nf%plant_n_uptake_flux
-      plant_p_uptake_flux          => col_pf%plant_p_uptake_flux
-
       ! set time steps
-      dt = real( get_step_size(), r8 )
-
-      call get_curr_date(yr, mon, day, sec)
-      if (spinup_state == 1 .and. yr .gt. nyears_ad_carbon_only) then 
-        call cnallocate_carbon_only_set(.false.)
+      if (spinup_state == 1 .and. yr .gt. nyears_ad_carbon_only) then
+         carbon_only = .false.
```

Comment:
if this is being set to false, do we also need to set CN or CNP to true? Or check them?

Reply:
I believe carbon_only is the only one adjusted mid-run after initialization based on the namelist.  Having that if statement being checked every time step is inelegant but I don't know what would have to change to get rid of that.